RemoteIDModule/BLE_TX.cpp

```diff
@@ -228,19 +228,22 @@ bool BLE_TX::transmit_legacy(ODID_UAS_Data &UAS_data)
         break;
 
     case  1:
-        ODID_BasicID_encoded basicid_encoded;
-        memset(&basicid_encoded, 0, sizeof(basicid_encoded));
-        if (encodeBasicIDMessage(&basicid_encoded, &UAS_data.BasicID[0]) != ODID_SUCCESS) {
-            break;
+        if (g.have_basic_id_info()) { 
```

Comment:
have_basic_id_info() is only true if the parameters are set. At this stage basic ID can also come from the GCS from the MissionPlanner plugin
I think to maintain that functionality we'd need to check if all 3 fields are value instead?

Reply:
>I think to maintain that functionality we'd need to check if all 3 fields are value instead?

Well see this PR https://github.com/ArduPilot/ArduRemoteID/pull/73

In my implementation, the first basic ID comes always from the parameters, The second basic ID can be from the parameters or from the GCS. You want to make the first basic ID tamper proof given the MoC. I don't see how you can do it in the GCS, because the user can edit this field. Entering the second ID in the GCS is useful in cases where you want to set a temporarily basic ID. So in https://github.com/ArduPilot/ArduRemoteID/pull/73, if a user sets a UAS_ID value, it will use the second basic ID (and overwrite the  one is set in the parameters). 

It (PR https://github.com/ArduPilot/ArduRemoteID/pull/73) is also backward compatible because if the UAS_ID is not set in the parameters, the second basic ID will be broadcast with the value from the GCS.

Reply:
> have_basic_id_info() is only true if the parameters are set. At this stage basic ID can also come from the GCS from the MissionPlanner plugin I think to maintain that functionality we'd need to check if all 3 fields are value instead?

Why do you want to remove line 232 - 234? You need those to encode the basic ID contents into a BT4 payload. 

PR https://github.com/ArduPilot/ArduRemoteID/pull/82 addresses your concern w.r.t. that all 3 fields have valid data. This is also addressed in PR https://github.com/ArduPilot/ArduRemoteID/pull/73.